sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -101,12 +101,16 @@
      * @throws RuntimeException If the directory is using a malformed URL.
      */
     public URL getDirectoryUrl() {
+        String directoryURLString = String.format("%s/%s/%s", azureFileStorageClient.getUrl(),
+            shareName, directoryPath);
+        if (snapshot != null) {
+            directoryURLString = String.format("%s?snapshot=%s", directoryURLString, snapshot);
```

Comment:
Instead of doing multiple String::format calls across if statements, why not use a string builder?

Reply:
Any benefits of StringBuilder?

Reply:
Made the changes.